services/user.py

```diff
@@ -0,0 +1,50 @@
+from django.contrib.auth import get_user_model
+from typing import Optional
+
+User = get_user_model()
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: Optional[str] = None, first_name: Optional[str] = None,
+    last_name: Optional[str] = None
+) -> User:
+    user = User.objects.create_user(
+        username=username,
+        password=password)
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+    user.save()

```

Comment:
You are calling the `save()` method twice here: once implicitly inside `create_user()` and once explicitly at line 22. According to the checklist, you should try to call the `save()` method only once. You can do this by using `create()` method instead of `create_user()` and setting the password with `set_password()` method. Note that `create_user()` also hashes the password, so you need to do it manually when using `create()`.